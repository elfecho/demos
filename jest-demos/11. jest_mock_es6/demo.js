import Util from './util';

export default () => {
  const util = new Util();
  util.a();
  util.b();
}