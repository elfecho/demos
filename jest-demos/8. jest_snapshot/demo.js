export const generateConfig = () => {
  return {
    server: 'http://localhost',
    port: 8080,
    domain: 'localhost',
    time: new Date(),
    id: Math.floor(Math.random() * 50)
  }
}

export const generateAnotherConfig = () => {
  return {
    server: 'http://localhost',
    port: 8080,
    domain: 'localhost2',
    time: new Date()
  }
}