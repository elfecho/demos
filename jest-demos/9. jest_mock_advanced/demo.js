import axios from 'axios';

export const fetchData = () => {
  return axios.get('/user.json').then(res => res.data);
}

export const getNumber = () => {
  return 123
}