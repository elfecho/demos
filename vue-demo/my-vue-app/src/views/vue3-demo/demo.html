<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>vue3.0 demo</title>
  <script src="https://unpkg.com/vue@next"></script>
</head>
<body>
  <div id="root"></div>
  <script>
    Vue.createApp({
      data(){
        return {
          content: '1'
        }
      },
      template: `<div>{{content}}</div>`
    }).mount('#root')
  </script>
</body>
</html>