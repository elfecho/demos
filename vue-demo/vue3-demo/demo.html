<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>vue3.0 demo</title>
  <script src="https://unpkg.com/vue@next"></script>
  <style>
    .area{
      position: absolute;
      left: 50%;
      top: 50%;
      transform: translate(-50%, -50%);
      width: 200px;
      height: 300px;
      background-color: #333;
      z-index: 88;
    }
    .mask {
      position: absolute;
      left: 0;
      right: 0;
      top: 0;
      bottom: 0;
      background-color: #000;
      opacity: 0.5;
    }
  </style>
</head>
<body>
  <div id="root"></div>
  <div id="hello"></div>
  <script>
    // 自定义指令 directive

    const app = Vue.createApp({
      // directives,
      data() {
        return {
          show: false,
          distance: 110
        }
      },
      methods: {
        handleBtnClick() {
          this.show = !this.show
        }
      },
      template: `
        <div>
          <div class="area">
            <button @click="handleBtnClick">按钮</button>
            <teleport to="#hello">
              <div class="mask" v-show="show"></div>
            </teleport>
          </div>
        </div>
      `
    })
    const vm = app.mount('#root')
  </script>
</body>
</html>