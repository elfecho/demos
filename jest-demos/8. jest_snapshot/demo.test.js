import { generateConfig, generateAnotherConfig } from "./demo";

test("测试 generateConfig 函数", () => {
  expect(generateConfig()).toMatchSnapshot({
    time: expect.any(Date),
    id: expect.any(Number)
  });
});

test("测试 generateAnotherConfig 函数", () => {
  expect(generateAnotherConfig()).toMatchSnapshot({
    time: expect.any(Date)
  });
});

// test("测试 generateConfig 函数", () => {
//   expect(generateConfig()).toMatchInlineSnapshot(
//     {
//       time: expect.any(Date)
//     },
//     `
//     Object {
//       "domain": "localhost",
//       "port": 8080,
//       "server": "http://localhost",
//       "time": Any<Date>,
//     }
//   `
//   );
// });

test("测试 config", () => {
  let config = {
    server: "http://localhost",
    port: 8080,
    domain: "localhost2"
  };
  expect(config).toMatchInlineSnapshot(`
    Object {
      "domain": "localhost2",
      "port": 8080,
      "server": "http://localhost",
    }
  `);
});
