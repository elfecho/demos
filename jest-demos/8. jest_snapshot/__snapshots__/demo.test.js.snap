// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 generateAnotherConfig 函数 1`] = `
Object {
  "domain": "localhost2",
  "port": 8080,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;

exports[`测试 generateConfig 函数 1`] = `
Object {
  "domain": "localhost",
  "id": Any<Number>,
  "port": 8080,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;
